internal/discover/symlinks.go

```diff
@@ -110,3 +132,70 @@ func (d additionalSymlinks) isDriverLibrary(libraryName string, filename string)
 	match, _ := filepath.Match(pattern, filename)
 	return match
 }
+
+func (d *additionalSymlinks) getDotSoSymlinks(libraryHostPath string, libraryContainerPath string) ([]string, error) {
+	libraryDir, libraryName := filepath.Split(libraryContainerPath)
+	if !d.isDriverLibrary("*", libraryName) {
+		return nil, nil
+	}
+
+	soname, err := getSoname(libraryHostPath)
+	if err != nil {
+		return nil, err
+	}
+
+	var soSymlinks []string
+	// Create the SONAME -> libraryName symlink.
+	if soname != libraryName {
+		s := Symlink{
+			target: libraryName,
+			link:   filepath.Join(libraryDir, soname),
+		}
+		soSymlinks = append(soSymlinks, s.String())
+	}
+
+	// Create the .so -> SONAME symlink
+	if soLink := getSoLink(soname); soLink != "" && soLink != soname {
+		s := Symlink{
+			target: soname,
+			link:   filepath.Join(libraryDir, soLink),
+		}
+		soSymlinks = append(soSymlinks, s.String())
+	}
```

Comment:
I still think this could be a bit more readable (in terms of the conditional cases). Maybe just a comment on what the cases are / mean would be sufficient, e.g.:
```
// Create the .so -> SONAME symlink
// If soname does not have a '.so' extension or soname itself ends with '.so'
// we skip this (because we will have just created the same symlink above).
```